.theme/app/components/Mermaid.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+  <div ref="mermaidContainer" class="mermaid">
+    <slot />
+  </div>
+</template>
+<script setup>
+import { useNuxtApp } from "nuxt/app";
+import { onMounted, ref, useSlots } from "vue";
+
+const { $mermaid } = useNuxtApp();
+const mermaidContainer = ref(null);
+const slot = useSlots().default;
+
```

Comment:
separate composable usage and field setup.

use nuxt app and use slots, then mermaid container and slot.